### ISSUE TITLE
Use octokit?

### ISSUE THREAD

**Issue by NPadrutt** [michelsalib/Gi7#74]
I just found this library:

https://github.com/octokit/octokit.net

It seems it provides an implementation for a lot who we also need. Therefore it sounds like a good idea to use the library and share development here.

**Comment by RandomlyKnighted** [comment]
It mentions Windows 8 but not Windows Phone. If we did decide to go with something like this we would definitely need to test it first to make sure it's compatible.

**Comment by NPadrutt** [comment]
sure. I saw in the Issue list that there are already windows phone apps out there who are using it. For example: 
http://www.windowsphone.com/de-de/store/app/octocentral/5450b4ca-875b-4d47-8d2a-75004a873f87
(Although this seems to be a WP8.1 Store App and not a 8.1 Silverlight App, but maybe this would also be a consideration).

**Comment by AlbertoMonteiro** [comment]
octokit os mantained by The github team, we could test it, I guess that the github windows app use it.

**Comment by NPadrutt** [comment]
I made a quick test. At least adding works without any problems.

**Comment by RandomlyKnighted** [comment]
I think it would definitely simplify the project in the long run. The problem I see with using it is there would be a good bit that would need to be changed since this would most likely remove the need for RestSharp.

**Comment by NPadrutt** [comment]
I think there should be transition time while the existing things can use the existing code and new things can use octokit. But your certainly right that this would mean bigger changes to the project. But IMHO it's worth it.

**Comment by RandomlyKnighted** [comment]
I think it's definitely worth the time. If we are going to do this, then we should probably make a list of what changes we would want to implement first using Octokit.

**Comment by AlbertoMonteiro** [comment]
Agree with @RandomlyKnighted

**Comment by NPadrutt** [comment]
Agreed. We could make "refactor with octokit"-issues. Or a wikipage.

**Comment by RandomlyKnighted** [comment]
Why not both? The wiki could also be used for notes that we find on the Octokit project.

**Comment by NPadrutt** [comment]
good point.

**Comment by RandomlyKnighted** [comment]
I added a "Refactor-With-Ocotokit" label.

@michelsalib can you add a wiki to the project please?

**Comment by AlbertoMonteiro** [comment]
When fixing some issue, I used Octokit, I really love it, much better than use RestSharp!!!
:octocat: Octokit use TPL(async,await) FTW!!!!

**Comment by NPadrutt** [comment]
Yeah it's really nice to use :)

**Comment by AlbertoMonteiro** [comment]
I think we should split views for each developer and refactor everything with Octokit. Since a lot things isn't working properly.

**Comment by NPadrutt** [comment]
split views for each developer? you mean that everyone takes responsibilty for refactoring some views?

I agree that we have to refactor everything or nearly everything with octokit. The wikipage would be nice to organize it. But for that we need @michelsalib and it seems he isn't very active atm.

**Comment by AlbertoMonteiro** [comment]
Yeah, everyone take some view and do the job.
About @michelsalib we could change the main repository to be some of us that are active. 
What do you think?

**Comment by NPadrutt** [comment]
yeah I tend to something like that. I think we should that discuss that in skype. Maybe a bit easier than with comments here

**Comment by AlbertoMonteiro** [comment]
I'm eating now when I back to work we can talk if you are available.

Enviado pelo meu Windows Phone
________________________________

**Comment by NPadrutt** [comment]
Enjoy your meal.
I will be available although only for chat I guess.
________________________________
Von: Alberto Monteiro<mailto:notifications@github.com>
Gesendet: ‎15.‎01.‎2015 16:49
An: michelsalib/Gi7<mailto:Gi7@noreply.github.com>
Cc: Nino Padrutt<mailto:nino.padrutt@hotmail.com>
Betreff: Re: [Gi7] Use octokit? (#74)

I'm eating now when I back to work we can talk if you are available.

Enviado pelo meu Windows Phone
________________________________

**Comment by michelsalib** [comment]
Hey guys, thanks for working on this. I don't really mind moving the repo. Though if we only need to activate the wiki I can gladly do it :)

**Comment by michelsalib** [comment]
Wiki is on!
